### PR TITLE
HDDS-5356. [SCM-HA] SCM start failed with PipelineNotFoundException

### PR DESCRIPTION
**Pull request by bshashikant**
## What changes were proposed in this pull request?
ignore pipeline not found exception during updatePipelineState

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5356

## How was this patch tested?
Manual test

### CODE REVIEW

**Review by mukul1987 — APPROVED** [review]

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java [hunk 1]
```diff
@@ -246,10 +246,10 @@ public void updatePipelineState(
       HddsProtos.PipelineID pipelineIDProto, HddsProtos.PipelineState newState)
       throws IOException {
     PipelineID pipelineID = PipelineID.getFromProtobuf(pipelineIDProto);
-    Pipeline.PipelineState oldState =
-        getPipeline(pipelineID).getPipelineState();
-    lock.writeLock().lock();
+    Pipeline.PipelineState oldState = null;
     try {
+      oldState = getPipeline(pipelineID).getPipelineState();
```
Comment (bharatviswa504) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java:251]: Do we need to call getPipeline also under lock?

**Review by bharatviswa504 — COMMENTED** [review]
LGTM.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java (hunk 1 above)
Comment (bshashikant) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java:251]: It should be fine as is , as it ignores the PipelineNotFoundException.

**Comment by JacksonYao287** [comment]
LGTM.

**Review by bharatviswa504 — APPROVED** [review]

**Pull request merged**